app/react/src/client/preview/render.tsx

```diff
@@ -45,19 +52,42 @@ class ErrorBoundary extends Component<{
 
 const Wrapper = FRAMEWORK_OPTIONS?.strictMode ? StrictMode : Fragment;
 
+const StoryContextContext = createContext<StoryContext | undefined>(undefined);
+
+export const applyDecorator = (story: StoryFn, decorator: DecoratorFunction) => {
+  const Story = story as FunctionComponent<StoryContext>;
+  // You cannot override the parameters key, it is fixed
+  const BoundStory = (
+    { parameters, ...innerStoryContext }: StoryContext | { parameters: {} } = { parameters: {} }
+  ) => {
+    return (
+      <StoryContextContext.Consumer>
+        {(storyContext) => <Story {...{ ...storyContext, ...innerStoryContext }} />}
+      </StoryContextContext.Consumer>
```

Comment:
I'm curious why we're not using `useContext` instead of render props.

Reply:
I can't quite remember, but my guess would be that `Context.Consumer` is an older API than `useContext`.